pkg/network/protocols/telemetry/metric.go

```diff
@@ -38,6 +38,18 @@ func (c *Counter) Add(v int64) {
 	c.value.Add(v)
 }
 
+func (c *Counter) UpdateCounterWithDifference(v int64) {
+	differenceValue := v - c.Get()
+	if v < 0 || differenceValue <= 0 {
+		// Counters are always monotonic, so we don't allow negative numbers. We
+		// could enforce this by using an unsigned type, but that would make the
+		// API a little bit more cumbersome to use.
+		return
+	}
+
+	c.value.Add(differenceValue)
```

Comment:
I don't think that code belongs here, and generally the function is out of place
the counter should not take care of that
Keep the implementation simple

Reply:
Removed it